### ISSUE TITLE
PowerShell is slow to close after showing a Windows forms dialog

### ISSUE THREAD

**Issue by MartinGC94** [PowerShell/PowerShell#18315]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

1. Open PowerShell in the standard conhost
2. Paste the following code into the console
3. Close the spawned window and the PowerShell console window and note the time it takes to close

```
Add-Type -AssemblyName System.Windows.Forms
$Form=[System.Windows.Forms.Form]::new()
$Form.ShowDialog()
```

On my system it takes about 5 seconds to close. This slowdown also happens if you have used `Get-Help -ShowWindow` or `Out-Gridview` in that session.  
This slowdown does not happen in Windows PowerShell or if you use the Windows terminal for PowerShell.

### Expected behavior

```console
The PowerShell window instantly closes when clicking on the X.
```

### Actual behavior

```console
The PowerShell window takes 5 seconds to close when clicking on the X after having used a GUI.
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.3.0-preview.8
PSEdition                      Core
GitCommitId                    7.3.0-preview.8
OS                             Microsoft Windows 10.0.19044
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by StevenBucher98** [comment]
Hmm I am unable to reproduce this bug on the same preview version. It closes instantly after creating the form. Can you try it on the latest rc release? and have you installed .NET separately somehow that might be causing this? I am suspect that this might be a .NET bug rather than PowerShell.

**Comment by MartinGC94** [comment]
It still happens on the latest RC release, here's a clip that demonstrates it:

https://user-images.githubusercontent.com/42123497/199095579-9f098d2e-cbfe-44e4-bf78-78e88f6632cf.mp4

Correct me if I'm wrong, but isn't PowerShell bundled with .NET? So my .NET installations shouldn't matter. Anyway, here's my installed software: 
![image](https://user-images.githubusercontent.com/42123497/199097118-823ef783-f5ae-4440-9f39-0d0b7b83c568.png)

I doubt it's system specific because it happens on all 3 computers I use daily and it has been going on for a while now, I just didn't get around to reporting it until recently.

**Comment by StevenBucher98** [comment]
Apologies I misunderstood the reproduction steps, I thought the form was taking long to close and not PowerShell, I can confirm it reproduces for me as well

**Comment by ptkls** [comment]
This issue is still reproducible on the newest version.

**Comment by kilasuit** [comment]
This seems to be fixed between 7.5.0-preview.1 & 7.5.0-preview.2 for me but still repro's in 7.4.1

**Comment by MartinGC94** [comment]
Same for me, I don't see it anymore on preview 2 of 7.5.0. I'll close the issue.
